app/models/user.rb

```diff
@@ -1,2 +1,49 @@
+# frozen_string_literal: true
+
 class User < ApplicationRecord
+  has_many :posts
+  attr_accessor :remember_token
+  before_create :generate_token, :downcase_email
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
+
+  def self.digest(string)
+    # rubocop:disable Style/MultilineTernaryOperator
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
```

Comment:
According to the [Odin Project's](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) specifications, you ought to use `Digest::SHA1.hexdigest` to encrypt.
Please make changes accordingly.

**IMPORTANT NOTE!!!**
I do understand why you may have thought to use `BCrypt::Engine` due to the `bcrypt` requirement but this requirement was so that you can utilize the `has_secure_password` method, not for the encryption of the generated token. Hope this info helps you 😄 

Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_create_requirement.png) for verification of this request. 🙂